### PR TITLE
Deduplication

### PR DESCRIPTION
**Pull request by jajik**
PR moves `loc_get_id` into `common.c`. This solution uses the id position within the corresponding structs (at the beginning), which can be a bit tricky. The alternative solution with `common.h` and macro controlled getter was explored as well, but it was way too complicated (we use both `make` and `cmake` + Windows behave differently than sane OSs).

I'm not sure we want to merge this – maybe it would be better to accept some duplication so I'm opening this as a draft.

### CODE REVIEW

**Review by jfclere — CHANGES_REQUESTED** [review]
Please put that in another PR.

File: native/include/node.h [hunk 1]
```diff
@@ -51,6 +51,16 @@ typedef struct mem mem_t;
 /* configuration of the node received from jboss cluster. */
 struct nodemess
 {
+    /* part updated in httpd */
+    int id;                  /* id in table and worker id */
```
Comment (jfclere) [inline native/include/node.h:55]: The comment needs to after the int id;
In fact please just move the int id;

**Review by jfclere — CHANGES_REQUESTED** [review]

File: native/include/node.h (hunk 1 above)
Comment (jajik) [reply native/include/node.h:55]: I wanted to preserve the structure within the struct, but as you wish, that's not a problem.

Comment (jajik) [reply native/include/node.h:55]: Done. I also added the note as I did in other cases.

**Comment by jajik** [comment]
(I've just rebased to the latest main.)

**Review by jfclere — APPROVED** [review]

**Pull request merged**